tests/sentry/models/test_releasefile.py

```diff
@@ -35,6 +35,20 @@ def test_normalize(self):
         # world, but worth documenting the behavior
         assert n("foo.js") == ["foo.js", "~foo.js"]
 
+    def test_count_artifacts(self):
+        assert self.release.count_artifacts() == 0
+        for count in (3, 1, None, 0):
+            file = self.create_file(name=f"dummy-{count}.txt")
+            ReleaseFile.objects.create(
+                file=file,
+                name=f"dummy-{count}.txt",
+                organization=self.organization,
+                release=self.release,
+                artifact_count=count,
+            )
+
+        assert self.release.count_artifacts() == 5
```

Comment:
Why is this 5? There are 4 files and the sum of the artifact_count value is 4.

Reply:
The logic is basically:

```python
if releasefile.artifact_count is None:
    # This a a legacy release file, count as one
    count = 1
else:
    count = releasefile.artifact_count
 ```